// @fireblocks/ts-sdk/dist/models/compliance-result-statuses-enum.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplianceResultStatusesEnum = void 0;
/**
 * Status of compliance result screening.
 * @export
 * @enum {string}
 */
exports.ComplianceResultStatusesEnum = {
    Started: 'Started',
    NetworkConnectionAddressResolve: 'NetworkConnectionAddressResolve',
    ScreeningPrepare: 'ScreeningPrepare',
    AmlStarted: 'AMLStarted',
    AmlCompleted: 'AMLCompleted',
    AmlFailed: 'AMLFailed',
    AmlinBackground: 'AMLInBackground',
    TrPreconditionChecks: 'TRPreconditionChecks',
    TrStarted: 'TRStarted',
    TrCompleted: 'TRCompleted',
    TrFailed: 'TRFailed',
    Completed: 'Completed',
    IncomingStarted: 'IncomingStarted',
    IncomingScreeningPrepare: 'IncomingScreeningPrepare',
    IncomingWaitForFirstConfirmation: 'IncomingWaitForFirstConfirmation',
    AmlIncomingStarted: 'AMLIncomingStarted',
    AmlIncomingCompleted: 'AMLIncomingCompleted',
    AmlIncomingFailed: 'AMLIncomingFailed',
    AmlIncomingInBackground: 'AMLIncomingInBackground',
    TrIncomingStarted: 'TRIncomingStarted',
    TrIncomingCompleted: 'TRIncomingCompleted',
    TrIncomingFailed: 'TRIncomingFailed',
    IncomingCompleted: 'IncomingCompleted'
};
