// @fireblocks/ts-sdk/dist/models/exchange-type.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExchangeType = void 0;
/**
 * Exchange account\'s type
 * @export
 * @enum {string}
 */
exports.ExchangeType = {
    IndependentReserve: 'INDEPENDENT_RESERVE',
    EnclaveMarkets: 'ENCLAVE_MARKETS',
    Bit: 'BIT',
    Coinflex: 'COINFLEX',
    Kucoin: 'KUCOIN',
    Pxs: 'PXS',
    Liquid: 'LIQUID',
    Bithumb: 'BITHUMB',
    Bitfinex: 'BITFINEX',
    Bitso: 'BITSO',
    Bitstamp: 'BITSTAMP',
    Kraken: 'KRAKEN',
    Krakenintl: 'KRAKENINTL',
    Binance: 'BINANCE',
    Binanceus: 'BINANCEUS',
    Cryptocom: 'CRYPTOCOM',
    BybitV2: 'BYBIT_V2',
    Coinbasepro: 'COINBASEPRO',
    Coinbaseprime: 'COINBASEPRIME',
    Coinbaseinternational: 'COINBASEINTERNATIONAL',
    Whitebit: 'WHITEBIT',
    Coinbaseexchange: 'COINBASEEXCHANGE',
    Korbit: 'KORBIT',
    Hitbtc: 'HITBTC',
    Gemini: 'GEMINI',
    Circle: 'CIRCLE',
    Bitmex: 'BITMEX',
    Huobi: 'HUOBI',
    Deribit: 'DERIBIT',
    OkcoinV5: 'OKCOIN_V5',
    Okex: 'OKEX',
    Coinmetro: 'COINMETRO',
    Gateio: 'GATEIO',
    Scrypt: 'SCRYPT',
    Coinhako: 'COINHAKO',
    Lightbit: 'LIGHTBIT',
    Bullish: 'BULLISH',
    CanvasConnect: 'CANVAS_CONNECT',
    Bitget: 'BITGET',
    Luno: 'LUNO',
    BitGenera: 'BIT_GENERA',
    Transfero: 'TRANSFERO'
};
