// @fireblocks/ts-sdk/dist/models/travel-rule-validate-transaction-response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Fireblocks API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.6.2
 * Contact: support@fireblocks.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TravelRuleValidateTransactionResponseAddressSourceEnum = exports.TravelRuleValidateTransactionResponseBeneficiaryAddressTypeEnum = void 0;
exports.TravelRuleValidateTransactionResponseBeneficiaryAddressTypeEnum = {
    Unknown: 'UNKNOWN',
    Hosted: 'HOSTED',
    Unhosted: 'UNHOSTED'
};
exports.TravelRuleValidateTransactionResponseAddressSourceEnum = {
    Unknown: 'UNKNOWN',
    AddressHash: 'ADDRESS_HASH',
    AddressGraph: 'ADDRESS_GRAPH',
    Chainalysis: 'CHAINALYSIS',
    Elliptic: 'ELLIPTIC',
    Crystal: 'CRYSTAL'
};
